zerver/views/auth.py

```diff
@@ -321,10 +321,16 @@ def finish_mobile_flow(request: HttpRequest, user_profile: UserProfile, otp: str
 
 def create_response_for_otp_flow(key: str, otp: str, user_profile: UserProfile,
                                  encrypted_key_field_name: str) -> HttpResponse:
+    
+    realm_uri = user_profile.realm.uri
+    # Check if the mobile URI is overridden in settings, if so replace it
+    if realm_uri in settings.REALM_MOBILE_URIS:
+        realm_uri = settings.REALM_MOBILE_URIS[realm_uri]
```

Comment:
This needs a test, probably just a variant of `test_social_auth_mobile_success` with `with self.settings(REALM_MOBILE_URIS=...)` used to prepare it; we don't merge anything to authentication code without tests :)

Reply:
The latest update should resolve this.